### PR TITLE
Enhance start script

### PR DESCRIPTION
**Pull request by Lucki**
The script should work like before, hopefully.
Changes are:
- use `$XDG_CACHE_HOME/truckermp-cli` with `$HOME/.cache/truckersmp-cli` fallback
- use existing local `truckersmp` if available for fallback
- possible to specify the mod path with `-d <path>`
- show help with `-h`
- `GAMEDIR` is optional to only update and validate the mod files
- catch wrong given `GAMEDIR`
- reduce the output of `wget`
- place temporary files in `/tmp` for the filelist and the checksums

I'm not sure about the reduced output of `wget` because it sometimes hangs forever because of the two download locations I have no clue of.

### CODE REVIEW

File: truckersmp-cli [hunk 1]
```diff
@@ -1,33 +1,75 @@
...
+        usage
+        quit 0
+        ;;
+    esac
+done
+
+shift $((OPTIND-1))
+[ "${1:-}" = "--" ] && shift
+
+if [ -z "$1" ]; then
```
Comment (lhark) [inline truckersmp-cli]: I feel like the default behaviour when called with no argument should be to display the usage message. Maybe you could add an option like -u for upgrade only ?

**Comment by JonathanP89** [comment]
How does this work now, it's saying the "Unable to find ETS2 or ATS in this GAMEDIR.", where do I find the game in steam?

**Comment by Lucki** [comment]
You need specify your windows game folder just like before.

You can download the files through the full steam client in wine/windows or you can use steamcmd with `@sSteamCmdForcePlatformType windows`.

**Comment by JonathanP89** [comment]
So, I did it like I was doing it before when it worked but this still comes up "Unable to find ETS2 or ATS in this GAMEDIR."

**Comment by Lucki** [comment]
What's the command you're using and can you please verify there's a `/bin/win_x64/eurotrucks2.exe` or `/bin/win_x64/amtrucks.exe` inside the given folder strukture.

**Comment by JonathanP89** [comment]
There is a bin/win_x64/eurotrucks2.exe and the command is ./truckersmp-cli ./truckersmp-cli /home/jon/.wine/drive_c/Program\ Files\ \(x86\)/Steam/steamapps/common/Euro\ Truck\ Simulator\ 2/

**Comment by Lucki** [comment]
You only need to call `truckersmp-cli` once. Can you try out `./truckersmp-cli /home/jon/.wine/drive_c/Program\ Files\ (x86)/Steam/steamapps/common/Euro\ Truck\ Simulator\ 2/`?

**Comment by JonathanP89** [comment]
Sorry, the command i posted is wrong, i use truckersmp-cli once

**Comment by Lucki** [comment]
No problem. My shell escapes `(` and `)`. Can you try `./truckersmp-cli /home/jon/.wine/drive_c/Program\ Files\ \(x86\)/Steam/steamapps/common/Euro\ Truck\ Simulator\ 2/`?

**Comment by JonathanP89** [comment]
Hm, for some reason I get this "unable to locate library (./truckersmp\core_ets2mp.dll)" and I did look in the truckersmp and couldn't find that file, I even redownloaded the zip

**Comment by Lucki** [comment]
Which zip?
This script should redownload all files on it's own. `-v` gives you a list of expected checksums of the files along with the files which are marked for download. You also can test this into a different folder by giving `-d ./path`

**Comment by lhark** [comment]
Sorry for the wait.

Everything looks good, I just reverted a few bashisms to stay as portable as possible.

**Comment by Lucki** [comment]
Great!

So in case @JonathanP89 still has problems please open a new issue.

**Pull request merged**